### ISSUE TITLE
Deprecating 'cls' in special matrix constructors

### ISSUE THREAD

**Issue by sylee957** [sympy/sympy#16187]
https://github.com/sympy/sympy/blob/9cb8736a8ab2495e8833de1f2a2b936f922f8111/sympy/matrices/common.py#L654-L903

> Also cls seems really awkward. When was it introduced? It seems like if someone wants a certain class, they should do something like TargetClass.jordon_block(size, eigenvalue) instead of passing in TargetClass to the cls kwarg.

> (And I agree about the cls option. I'm not sure what merit there is in passing this rather than using it as the calling class -- either way you have to type it.)

There were some discussions about the keyword argument, and I also think that the keyword would not be any useful at all.

**Comment by asmeurer** [comment]
Here is the original comment for reference https://github.com/sympy/sympy/pull/16102#issuecomment-470348586

**Comment by asmeurer** [comment]
I agree. The cls argument (the first argument of the classmethod) should be the class that is used. That way follows standard Python inheritance patterns.
